### PR TITLE
[#2496] 504s when opening project editor for big projects

### PR DESCRIPTION
**Pull request by punchagan**
- [ ] Test plan | Unit test | Integration test
- [ ] Copyright header
- [ ] Code formatting
- [ ] Documentation
- [ ] Change log entry

### CODE REVIEW

**Comment by punchagan** [comment]
Closes #2496

File: akvo/rsr/tests/test_templatetags.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+For additional details on the GNU license please see < http://www.gnu.org/licenses/agpl.html >.
+"""
+
+from django.test import TestCase
+
+from akvo.rsr.models import ProjectEditorValidation, ProjectEditorValidationSet
+from akvo.rsr.templatetags.project_editor import mandatory_or_hidden
+
+
+class TemplateTagssTestCase(TestCase):
```
Comment (zzgvh) [inline akvo/rsr/tests/test_templatetags.py]: Snake-speach mis-spell 😉

File: akvo/rsr/tests/test_templatetags.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    """Tests for template tags."""
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (zzgvh) [inline akvo/rsr/tests/test_templatetags.py]: It woule be nice to use the actions defined in ProjectEditorValidation here, something like:
```python
def setUpClass(cls):
    MANDATORY = ProjectEditorValidation.MANDATORY_ACTION
    HIDDEN = ProjectEditorValidation.HIDDEN_ACTION
    validation1 = [
        ('rsr_project.title', MANDATORY)
        ('rsr_project.subtitle', MANDATORY),
        ...
```

**Comment by punchagan** [comment]
Closed in favor of #2503

**Pull request closed without merging**